tests/test_utils.py

```diff
@@ -2766,6 +2766,96 @@ def _local_minimum_grid_heuristic_fn(state: S) -> float:
         ]
 
 
+def test_run_policy_guided_astar():
+    """Tests for run_policy_guided_astar()."""
+    S = Tuple[int, int]  # grid (row, col)
+    A = str  # up, down, left, right
+
+    arrival_costs = np.array([
+        [1, 1, 100, 1, 1],
+        [1, 100, 1, 1, 1],
+        [1, 100, 1, 1, 1],
+        [1, 1, 1, 100, 1],
+        [1, 1, 100, 1, 1],
+    ],
+                             dtype=float)
+
+    act_to_delta = {
+        "up": (-1, 0),
+        "down": (1, 0),
+        "left": (0, -1),
+        "right": (0, 1),
+    }
+
+    def _get_valid_actions(state: S) -> Iterator[Tuple[A, float]]:
+        r, c = state
+        for act in sorted(act_to_delta):
+            dr, dc = act_to_delta[act]
+            new_r, new_c = r + dr, c + dc
+            # Check if in bounds
+            if not (0 <= new_r < arrival_costs.shape[0] and \
+                    0 <= new_c < arrival_costs.shape[1]):
+                continue
+            yield (act, arrival_costs[new_r, new_c])
+
+    def _get_next_state(state: S, action: A) -> S:
+        r, c = state
+        dr, dc = act_to_delta[action]
+        return (r + dr, c + dc)
+
+    def _grid_check_goal_fn(state: S) -> bool:
+        # Bottom right corner of grid
+        return state == (4, 4)
+
+    def _grid_heuristic_fn(state: S) -> float:
+        # Manhattan distance
+        return float(abs(state[0] - 4) + abs(state[1] - 4))
+
+    def _policy(state: S) -> A:
+        # Move right until we can't anymore.
+        _, c = state
+        if c >= arrival_costs.shape[1] - 1:
+            return None
+        return "right"
+
+    initial_state = (0, 0)
+    num_rollout_steps = 10
+
+    # The policy should bias toward the path that moves all the way right, then
```

Comment:
i tried this out. I had to change the arrival cost for the bottom-middle state from 100 to 1 because otherwise the search wanted to enter that state from the right with the policy (to avoid the environment cost). but after that change, the result is
```
ipdb> action_sequence
['right', 'right', 'down', 'down', 'down', 'down']
```
as you expected